docs/modules/generic_send.md

```diff
@@ -0,0 +1,77 @@
+# Generic Send (generic_send.py)
+
+## Description
+
+This script is used to send data generated by the [PerunDataGenerator](PerunDataGenerator.md) to the Perun API. It is
+used by the service-specific sending scripts written in python.
+First it checks the input parameters, and then it runs the complete generic sending logic.
+Based on the number of parameters sent to the script, it determines the destination type and calls the `send` method.
+
+### `send` method
+
+Sends data generated for the specified service/facility to the desired destination.
+Destination type can be specified as the last parameter, default is 'host'.
+Unsupported destination types are **email** and **service-specific**.
+Then it prepares the transport command and calls the `process` method, results are printed to STDOUT/STDERR.
+
+Parameters:
+
+- `service_name`: name of the service to send data for
+- `facility_name`: name of the facility to send data for
+- `destination`: name of the destination to send data to
+- `destination_type`: optional destination type, default is 'host'
+- `opts`: list of additional options passed to the SSH/CURL transport command
+
+Based on the return codes of the `process` method
+
+- 124 - timeout
+- 0 - success
+- &lt;other&gt; - call was successful, but there was an error on the destination side
+
+### `prepare_url_transport_command` method
+
+Prepares base transport command for CURL connection. If available, adds common perun certificate.
+
+Parameters:
+
+- `temp`: temporary file
```

Comment:
i feel like these parameters are not much interesting, I can after all see them in the code.
i would prefer a high-level description of how the sender works as a whole, what are the capabilities (destination types), what needs to be configured for which use case (credentials, credentials), what data can it transfer (any generated by gen script) etc.

Reply:
maybe sections for each destination type would be suitable